src/main/java/com/box/sdk/BoxAPIConnection.java

```diff
@@ -123,6 +140,46 @@ public static BoxAPIConnection restore(String clientID, String clientSecret, Str
     }
 
     /**
+     * Authenticates the API connection for Box Developer Edition.
+     * @param entityId           An enterprise ID or a user ID.
+     * @param entityType         "enterprise" or "user", corresponding to entityId.
+     * @param privateKey         the private key corresponding to the public key configured with Box Developer Edition.
+     * @param privateKeyPassword the password for the private key.
+     * @throws JoseException
+     * @throws IOException
+     */
+    public void authenticate(String entityId, String entityType, String privateKey, String privateKeyPassword)
+        throws JoseException, IOException {
```

Comment:
I'd prefer that these exceptions be wrapped with a `BoxAPIException` for consistency. This way we don't have to expose classes from dependencies through the SDK.